Content.Client/Examine/ExamineSystem.cs

```diff
@@ -133,69 +188,151 @@ public async void DoExamine(EntityUid entity)
+        /// <summary>
+        ///     Fills the examine tooltip with a message and buttons if applicable.
+        /// </summary>
+        public void UpdateTooltipInfo(EntityUid player, EntityUid target, FormattedMessage message, bool getVerbs = true)
+        {
+            var vBox = _examineTooltipOpen?.GetChild(0).GetChild(0);
+            if (vBox == null)
             {
-                message = GetExamineText(entity, _playerManager.LocalPlayer?.ControlledEntity);
+                return;
             }
-            else
-            {
-                // Ask server for extra examine info.
-                RaiseNetworkEvent(new ExamineSystemMessages.RequestExamineInfoMessage(entity));
 
-                ExamineSystemMessages.ExamineInfoResponseMessage response;
-                try
-                {
-                    _requestCancelTokenSource = new CancellationTokenSource();
-                    response =
-                        await AwaitNetworkEvent<ExamineSystemMessages.ExamineInfoResponseMessage>(_requestCancelTokenSource
-                            .Token);
-                }
-                catch (TaskCanceledException)
-                {
-                    return;
-                }
-                finally
-                {
-                    _requestCancelTokenSource = null;
-                }
-
-                message = response.Message;
+            SortedSet<Verb>? set = null;
+            if (getVerbs)
+            {
+                // Get verbs
+                var verbs = _verbSystem.GetVerbs(target, player, VerbType.Examine);
+                verbs.TryGetValue(VerbType.Examine, out set);
             }
 
             foreach (var msg in message.Tags.OfType<FormattedMessage.TagText>())
             {
                 if (string.IsNullOrWhiteSpace(msg.Text)) continue;
 
-                var richLabel = new RichTextLabel();
+                var richLabel = new RichTextLabel() { Margin = new Thickness(4, 4, 0, 4)};
                 richLabel.SetMessage(message);
                 vBox.AddChild(richLabel);
                 break;
             }
+
+            if (set != null)
+            {
+                AddVerbsToTooltip(set);
+            }
+        }
+
+        private void AddVerbsToTooltip(IEnumerable<Verb> verbs)
+        {
+            if (_examineTooltipOpen == null)
+                return;
+
+            var buttonsHBox = new BoxContainer
+            {
+                Name = "ExamineButtonsHBox",
+                Orientation = LayoutOrientation.Horizontal,
+                HorizontalAlignment = Control.HAlignment.Right,
+                VerticalAlignment = Control.VAlignment.Bottom,
+            };
+
+            // Examine button time
+            foreach (var verb in verbs)
+            {
+                if (verb.IconTexture == null)
+                    continue;
+
+                // snowflake this for now.
+                if (verb.Text == "Basic")
+                    continue;
```

Comment:
this is sus but can't think of a better way to exclude it

Reply:
I guess this should really be a Loc.GetString("examine-verb-name")

And maybe the `"examine-verb-name"` should itself be a const string somewhere, so that if ever anyone updates `AddExamineVerb()` this won't accidentally be forgotten. Then its still a janky solution, but at least slightly more robust.

Reply:
I guess there should be a better way of attaching extra data to verbs, especially if they get used in more places like is being done here (e.g. if they eventually need data for the action toolbar or something).

I guess you could create an `ExamineVerb` class that inherits from `Verb` that just has a `ShowInTooltip` bool? But then you have this confusing combination of an Examine VerbType and a class.

Going off topic here, but  the way verb types also work at the moment is a bity janky. The type is not actually a property of the verb, but has to be passed alongside the verb wherever it is used. E.g. for:
```cs
_verbSystem.ExecuteVerb(_examinedEntity, button.Verb, VerbType.Examine);
```
Initially I did just have verb types be classes that inherit from verbs, but IIRC I scrapped it when I realised generic event subscriptions like `GetVerbsEvent<TVerb> where TVerb : Verb` weren't a thing you could do. But maybe its worth going back to that and just getting rid of the verb-type struct? Then this issue of needing examine-tooptip specific data for examine0verbs would be easy to solve,

Reply:
Actually I don't even remember why the generic event stuff didn't work. Testing again now it just.... works?